File: src/test/resources/collections/List2.java
package collections;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;

public class List2 {

    public static void main(String[] args) {
        List<String> al = new ArrayList<>();
        al.add("1");
        al.add("2");
        al.add("3");
        al.remove("2");
        al.remove(1);
        al.addAll(Arrays.asList("5", "6", "7"));

        System.out.println(al.size());
        System.out.println(al.isEmpty());
        System.out.println(al.contains("3"));
        System.out.println(al.toArray());
        System.out.println(al.toArray(new String[al.size()]));

        LinkedList<String> ll = new LinkedList<>();
        ll.addFirst("f1");
        ll.addLast("l1");
        ll.removeFirst();
        ll.removeLast();
        ll.remove("1");
        System.out.println(ll);

    }

}

File: src/test/resources/collections/List2.cs
namespace collections {
    public class List2 {
        public static void Main(string[] args) {
            System.Collections.Generic.IList<string> al = new System.Collections.Generic.List<string>();
            al.Add("1");
            al.Add("2");
            al.Add("3");
            al.Remove("2");
            al.RemoveAtReturningValue(1);
            Sharpen.Collections.AddAll(al, Sharpen.Arrays.AsList("5", "6", "7"));
            al[0] = "1";
            System.Console.Out.WriteLine(al.Count);
            System.Console.Out.WriteLine((al.Count == 0));
            System.Console.Out.WriteLine(al.Contains("3"));
            System.Console.Out.WriteLine(Sharpen.Collections.ToArray(al));
            System.Console.Out.WriteLine(Sharpen.Collections.ToArray(al, new string[al.Count]));
            System.Collections.Generic.LinkedList<string> ll = new System.Collections.Generic.LinkedList<string>();
            ll.AddFirst("f1");
            ll.AddLast("l1");
            Sharpen.Collections.RemoveFirst(ll);
            Sharpen.Collections.RemoveLast(ll);
            ll.Remove("1");
            System.Console.Out.WriteLine(ll);
        }
    }
}